core/src/test/java/org/infinispan/container/versioning/AbstractClusteredWriteSkewTest.java

```diff
@@ -301,19 +301,22 @@ private void doTestWriteSkewWithPassivation(boolean executeOnPrimaryOwner, Opera
       }
 
       DataContainer dataContainer = TestingUtil.extractComponent(primaryOwner, DataContainer.class);
-      MagicKey[] keys = new MagicKey[MAX_ENTRIES];
+      int retryCount = 5;
       int insertCount = 10;
-      for (int i = 0; i < insertCount; ++i) {
-         for (int j = 0; j < MAX_ENTRIES; ++j) {
-            MagicKey tempKey = keys[j];
-            if (tempKey == null) {
-               tempKey = new MagicKey("other-key-" + j, primaryOwner);
-               keys[j] = tempKey;
+      for (int retry = 0; retry < 5; ++retry) {
```

Comment:
That was the goal, unfortunately there is no 'peak' equivalent for caffeine.  So this could cause it to fail possibly.  That was my initial goal actually (hence retryCount variable name).  It should be named something else now :P